addon-test-support/@ember/test-helpers/setup-rendering-context.js

```diff
@@ -78,95 +82,66 @@ export default function(context) {
-            run(toplevelView, 'appendTo', '#ember-testing');
-            hasRendered = true;
-          }
-
-          // using next here because the actual rendering does not happen until
-          // the renderer detects it is dirty (which happens on backburner's end
-          // hook), see the following implementation details:
-          //
-          // * [view:outlet](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/views/outlet.js#L129-L145) manually dirties its own tag upon `setOutletState`
-          // * [backburner's custom end hook](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/renderer.js#L145-L159) detects that the current revision of the root is no longer the latest, and triggers a new rendering transaction
-          return nextTickPromise();
-        })
-        .then(() => {
-          // ensure the element is based on the wrapping toplevel view
-          // Ember still wraps the main application template with a
-          // normal tagged view
-          //
-          // In older Ember versions (2.4) the element itself is not stable,
-          // and therefore we cannot update the `this.element` until after the
-          // rendering is completed
-          element = document.querySelector('#ember-testing > .ember-view');
-
-          return settled();
-        });
-    };
+        stateToRender.name = 'index';
+        outletState.outlets.main = { render: stateToRender, outlets: {} };
 
-    Object.defineProperty(context, 'element', {
-      enumerable: true,
-      configurable: true,
-      get() {
-        return element;
-      },
-    });
+        toplevelView.setOutletState(outletState);
+
+        // using next here because the actual rendering does not happen until
+        // the renderer detects it is dirty (which happens on backburner's end
+        // hook), see the following implementation details:
+        //
+        // * [view:outlet](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/views/outlet.js#L129-L145) manually dirties its own tag upon `setOutletState`
+        // * [backburner's custom end hook](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/renderer.js#L145-L159) detects that the current revision of the root is no longer the latest, and triggers a new rendering transaction
+        return settled();
+      });
+    };
 
     if (global.jQuery) {
       context.$ = function $(selector) {
         // emulates Ember internal behavor of `this.$` in a component
         // https://github.com/emberjs/ember.js/blob/v2.5.1/packages/ember-views/lib/views/states/has_element.js#L18
-        return selector ? global.jQuery(selector, element) : global.jQuery(element);
+        return selector ? global.jQuery(selector, context.element) : global.jQuery(context.element);
```

Comment:
Unrelated, but I just learned that `context.element` is a jQuery element if you have jQuery. Is that intended?
I'd have expeted user to do the wrapping themselves if they want to, like

```
import $ from 'jquery';
// ...
  assert.equal($(this.element).hasClass('foo'));
```

Reply:
@cibernox `this.$ = $(this.element)`, `this.element` is always a native DOM element

Reply:
@cibernox - That is very surprising indeed! As you can see (just above in the diff) this is how we set `context.element` now (was the same before but done via `Object.defineProperty`):

```js
    context.element = document.querySelector('#ember-testing > .ember-view');
```

How could that _ever_ be a jQuery element? Are you certain that this wasn't in a `moduleForComponent` test?

Reply:
Nevermind. I read this ternary completely wrong. I thought the line was setting `context.element` to `global.jQuery(context.element)`.